### ISSUE TITLE
feat:RedditSync - Path /u/ requests to use /user/ endpoint

### ISSUE THREAD

**Issue by garpunkal** [ReVanced/revanced-patches#3634]
### Feature description

(https://www.reddit.com/user/username/) endpoint in favor of /user/username and "fixed" it for old.reddit.com by adding a redirect for that, but not the other things under it. Critically:

https://reddit.com/u/spez/about.json (500 error)

vs

https://reddit.com/user/spez/about.json (works fine)

### Motivation

Currently the About section in sync is broken for users because the endpoint provides a 500 error. 

It's important because sync breaks on the user profile page.

### Acknowledgements

- [X] I have checked all open and closed feature requests and this is not a duplicate
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by trulow** [comment]
Can I offer a $50 USD bounty for this to be patched?

**Comment by oSumAtrIX** [comment]
Claiming

**Comment by trulow** [comment]
@oSumAtrIX 

I DM'd you on reddit. Once the fix is live, send me your contact info to send the bounty.

**Comment by oSumAtrIX** [comment]
Here's the patch in action:

https://github.com/user-attachments/assets/0d035b65-0dfd-4fa5-9d2c-8a3af2af063a

**Comment by DVDAndroid** [comment]
I just checked this issue and my unpatched app and is working fine.
Do you think that this patch will be still needed?

**Comment by elmagio** [comment]
> I just checked this issue and my unpatched app and is working fine. Do you think that this patch will be still needed?

Huh that's weird, it also started working again for me, without a patch. Wasn't working as of an hour ago.

**Comment by oSumAtrIX** [comment]
It was a known bug on Reddits end confirmed by Reddit devs a couple days ago. It was bound to be fixed like the other login bugs caused by Reddit in the past. (The server returned 500, an internal error)

**Comment by trulow** [comment]
Does you new patch have a check to see if the app receives a 500 error, to then use /user instead?

If it does, maybe we can still implement it as a backup.

**Comment by oSumAtrIX** [comment]
No, it simply changes the endpoint to /user. I think it's even the "new" endpoint and /u/ is deprecated but exists for backwards compatibility reasons. Should me mentioned in the docs

**Comment by trulow** [comment]
I just went over the API docs, and it seems like @oSumAtrIX is correct.

I think we should implement this patch as the [API docs](https://www.reddit.com/dev/api/#GET_api_multi_user_{username}) only reference /user/ and not /u/

My thinking on this is that it's only a matter of time until /u/ may break again causing another fix in the future.

**Comment by oSumAtrIX** [comment]
It won't make a difference. u/ wont break unless Reddit decides to drop backwards compatibility support at which point dozens of other apis would be broken and would need heavy fixing

**Comment by garpunkal** [comment]
I think we should have the patch available. If /user/ is the canonical URL then it gives the option to patch for future use.

**Comment by oSumAtrIX** [comment]
I'll keep it for now as it doesn't break anything but maybe the patch should be renamed to what it does, as it's called a fix whereas it doesn't fix current issue
